scipy/optimize/tests/test_optimize.py

```diff
@@ -1238,6 +1240,15 @@ def test_respect_maxiter(self, method):
         if method == 'SLSQP':
             assert sol.status == 9  # Iteration limit reached
 
+    @pytest.mark.parametrize('method', ['Nelder-Mead', 'Powell'])
```

Comment:
Are these the only two methods affected by this PR? If others are affected, consider testing them, too.

Reply:
Thanks. I missed 'fmin', 'fmin_powell', `brute`, so I added tests for these.